source/Cosmos.System2/Graphics/Fonts/ASC16.cs

```diff
@@ -0,0 +1,43 @@
+        static MemoryStream ASC16FontMS = new MemoryStream(Convert.FromBase64String(ASC16Base64));
+
+        //Note: the size of font is 8*16
+        
+        public static void DrawACSIIString(this Canvas canvas, Pen pen, string s, int x, int y)
```

Comment:
The two differences between this method and the DrawString in https://github.com/CosmosOS/Cosmos/blob/1e4cbb65e83a800666aba4ca8e244e7585e76594/source/Cosmos.System2/Graphics/Canvas.cs#L964, I can see is that this one supports multiline strings and that the other supports more than one font. So you could you please add multiline support to the other implementation? Or is there some advantage to this one, which I am not seeing.

Reply:
@quajak we can draw converted font from windows

Reply:
i think pcsfont are too old and nobody will make new font..

Reply:
Ah, what font format is it? How do you convert it? Just want to allow the user to also be able to create and load their own fonts.

Reply:
DZK

Reply:
actually this file has no a exactly format. but there's a lot tool to generate this files.

Reply:
![QQ截图20210121193216](https://user-images.githubusercontent.com/54205437/105345585-68631e80-5c1f-11eb-881e-2af451f8bb73.png)
  like this

Reply:
this format of fonts always used for monito or MCU

Reply:
In that case, I would prefer if you add support for this font format by adding a class inheriting from Font, similar to PCScreenFont, so that users can also load custom fonts from this format. This would allow both types of fonts to be used by the method in canvas. If you also add multiline support to the method in Canvas, then we have everything offered from the new method, without having two seperate methods to confuse users.

Reply:
![QQ截图20210121204220](https://user-images.githubusercontent.com/54205437/105352874-761da180-5c29-11eb-9c38-b82a23facbaa.png)
![QQ截图20210121204354](https://user-images.githubusercontent.com/54205437/105352883-79189200-5c29-11eb-89c8-c6785b7e52d8.png)